loadtest/loadtest.go

```diff
@@ -252,6 +253,18 @@ func New(config *Config, nc NewController, log *mlog.Logger) (*LoadTester, error
 		return nil, fmt.Errorf("could not validate configuration: %w", err)
 	}
 
+	var rlimit syscall.Rlimit
+	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
+		return nil, fmt.Errorf("could not get the Rlimit value: %w", err)
+	}
+
+	if maxActiveUsers := config.UsersConfiguration.MaxActiveUsers; uint64(
+		maxActiveUsers+(maxActiveUsers/4)) > rlimit.Max {
```

Comment:
Let's refactor 4 to be a constant. And use that here and in the api/agent.go file. Perhaps `loadFactor` is an apt name, although it's a bit hazy. @streamer45 - any thoughts on this?

Reply:
Maybe `connFactor` since the result is about connections. We might also want to either export it so we can use it in `NewControllerWrapper` or create a helper function (e.g. `MaxHTTPConns(maxUsers int) int`)  that given `MaxActiveUsers` returns the max connections value. This way we could also take care of some edge cases (e.g. input value too low).

Reply:
so is a helper function required for now? or just a constant will do?

Reply:
A helper function sounds like a good idea. :+1: